es/es-server/src/main/java/org/elasticsearch/common/lucene/search/Queries.java

```diff
@@ -56,9 +61,17 @@ public static Query newLenientFieldQuery(String field, RuntimeException e) {
 
     /**
      * Creates a new non-nested docs query
+     * @param indexVersionCreated the index version created since newer indices can identify a parent field more efficiently
      */
-    public static Query newNonNestedFilter() {
-        return new DocValuesFieldExistsQuery(SeqNoFieldMapper.PRIMARY_TERM_NAME);
+    public static Query newNonNestedFilter(Version indexVersionCreated) {
+        if (indexVersionCreated.onOrAfter(Version.ES_V_6_1_4)) {
+            return new DocValuesFieldExistsQuery(SeqNoFieldMapper.PRIMARY_TERM_NAME);
+        } else {
+            return new BooleanQuery.Builder()
+                .add(new MatchAllDocsQuery(), Occur.FILTER)
+//                .add(newNestedFilter(), Occur.MUST_NOT)
+                .build();
+        }
```

Comment:
why is this needed? looks incomplete and can be maybe skipped/removed as we do not support nested fields.